server/src/main/java/org/elasticsearch/script/AbstractFieldScript.java

```diff
@@ -111,6 +114,14 @@ protected final void emitFromSource() {
         }
     }
 
+    protected final void emitFromSourcePaths() {
```

Comment:
It's probably worth adding javadoc to explain how this is different from `emitFromSource`.

Reply:
Agreed, at a minimum! Questions from the previous comment are relevant here too.